lib/reversetunnel/transport.go

```diff
@@ -297,8 +298,16 @@ func (p *transport) start() {
 				return
 			}
 
-			p.log.Debug("Handing off connection to a local SSH service")
-			p.server.HandleConnection(utils.NewChConn(p.sconn, p.channel))
+			p.log.Debugf("Handing off connection to a local %q service.", dreq.ConnType)
+			switch dreq.ConnType {
+			case types.AppTunnel:
+				// Use channel connection wrapper that supports deadlines when
+				// proxying applications. This is required for websockets to
+				// work over Application Access.
+				p.server.HandleConnection(utils.NewCancelableChConn(p.sconn, p.channel))
+			default:
+				p.server.HandleConnection(utils.NewChConn(p.sconn, p.channel))
+			}
```

Comment:
Any reason why you separated these two and created `CancelableChConn` instead of just adding deadline support to `ChConn`?

Reply:
@russjones No good reason other than out of abundance of caution. I can just update ChConn if you feel strongly about it, but I just felt that this functionality is only required for app access and wanted to avoid the risk of potentially introducing some side-effects to other types of services like ssh (and esp. since it's going to a minor release).

Reply:
Makes sense for 6.1. What do you think updating `ChConn` on master to support deadlines but keeping them separate on `branch/v6`?

Reply:
@russjones Yeah, that was one of my alternative ideas as well. We can do that. I'll rebase this PR off of `branch/v6` and open the one to master with just updated ChConn.

Reply:
Alright, looks good to me, merge it in after rebase off of `branch/v6`.